### PR TITLE
Fixes requirement path and class references, typos and added .travis.yml

### PR DESCRIPTION
**Pull request by zanderwar**
Working towards #1 and #5

### CODE REVIEW

File: src/BBCodeParser.php [hunk 1]
```diff
@@ -24,7 +24,7 @@ class BBCodeParser extends TextParser {
 	 * @config
 	 * @var Boolean Set whether similies :), :(, :P are converted to images
 	 */
-	private static $allow_similies = false;
+	private static $allow_smilies = false;
```
Comment (zanderwar) [inline src/BBCodeParser.php]: This doesn't show that the references were also refactored

File: src/BBCodeParser.php [hunk 2]
```diff
@@ -126,7 +126,7 @@ public function parse() {
 		$config = parse_ini_file('BBCodeParser.ini', true);
 		$options = $config['SSHTMLBBCodeParser'];
 
-		require_once(__DIR__ . '/../thirdparty/HTML_BBCodeParser2/BBCodeParser2.php');
```
Comment (dhensby) [inline src/BBCodeParser.php]: Can this package be included by composer now?

File: src/BBCodeParser.php (hunk 1 above)
Comment (dhensby) [inline src/BBCodeParser.php]: Where are those references?

Comment (zanderwar) [inline src/BBCodeParser.php]: Only [here](https://github.com/zanderwar/silverstripe-bbcodeparser/blob/1755da9fadfbf556fa7d06da8ebad8d8c075f836/src/BBCodeParser.php#L135)

File: src/BBCodeParser.php (hunk 2 above)
Comment (zanderwar) [inline src/BBCodeParser.php]: Not by the look of it, it's a PEAR package

Comment (dhensby) [inline src/BBCodeParser.php]: looks like it is: https://github.com/pear/HTML_BBCodeParser2/blob/master/composer.json#L24

File: src/BBCodeParser.php (hunk 1 above)
Comment (dhensby) [inline src/BBCodeParser.php]: ok, well `similies_location` is also misspelt

**Comment by dhensby** [comment]
we also need a `.travis.yml` so we can run tests

File: src/BBCodeParser.php (hunk 2 above)
Comment (zanderwar) [inline src/BBCodeParser.php]: I'll PR it in separately if that's ok? for now I just needed it working (and it is) for forum upgrade and it's going to be replaced anyway

**Comment by zanderwar** [comment]
Sure, @dhensby if it's ok with you, i'll PR that in separately

File: src/BBCodeParser.php (hunk 1 above)
Comment (zanderwar) [inline src/BBCodeParser.php]: Hmm, where? I see it set [correctly](https://github.com/zanderwar/silverstripe-bbcodeparser/blob/1755da9fadfbf556fa7d06da8ebad8d8c075f836/src/BBCodeParser.php#L36)

Comment (dhensby) [inline src/BBCodeParser.php]: Ok. I'm just confused as to why we want to fix this typo and not the others?

File: src/BBCodeParser.php (hunk 2 above)
Comment (dhensby) [inline src/BBCodeParser.php]: Ok

**Comment by dhensby** [comment]
I'd like the travis config so we can have some certainty that this is working... There are lots of example ones you can use from modules that have already had the 4.0 upgrade done

**Comment by zanderwar** [comment]
Sure I'll get it done for you, I had not already as I didn't see the point in bringing to life something that is EOL/Archived it only made sense to fix the bugs preventing it from working in SS4

File: src/BBCodeParser.php (hunk 1 above)
Comment (zanderwar) [inline src/BBCodeParser.php]: If you meant the PHPDoc, that's been fixed, beyond that I searched the module globally for `simili*` and found nothing

**Comment by zanderwar** [comment]
Should the namespace change since it's no longer apart of framework?

**Comment by dhensby** [comment]
> Should the namespace change since it's no longer apart of framework?

Which namespace?

**Comment by zanderwar** [comment]
> Which namespace?

`namespace SilverStripe\View\Parsers;`

**Comment by dhensby** [comment]
erm - good point - I don't think we have to, though don't we need to set up the PSR loader settings in composer.json?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      env: DB=SQLITE CORE_RELEASE=4
+
+before_script:
+ - composer self-update || true
+ - git clone git://github.com/silverstripe/silverstripe-travis-support.git ~/travis-support
+ - php ~/travis-support/travis_setup.php --source `pwd` --target ~/builds/ss
+ - cd ~/builds/ss
+
+script:
+  - vendor/bin/phpunit "$COVERAGE" forum/tests
```
Comment (dhensby) [inline .travis.yml]: I think you want `vendor/bin/phpunit silverstripe-bbcodeparser/tests`

**Comment by zanderwar** [comment]
Finally 👍

**Comment by dhensby** [comment]
@sminnee do you have a view on the namespace?

**Comment by dhensby** [comment]
@zanderwar one last thing, do we need an `.upgrade.yml`?

**Comment by zanderwar** [comment]
> @zanderwar one last thing, do we need an .upgrade.yml?

If you mean get rid of it (because it already exists #2) then that's your call

**Comment by dhensby** [comment]
Ah. No. We want it.

**Comment by dhensby** [comment]
We can change namespaces later if needs be

**Pull request merged**